lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -444,7 +445,8 @@ std::error_code ImplicitSerializedModuleLoader::findModuleFilesInDirectory(
           (!ModuleBuffer && !ModuleDocBuffer)) &&
          "Module and Module Doc buffer must both be initialized or NULL");
 
-  if (LoadMode == ModuleLoadingMode::OnlyInterface)
+  if (LoadMode == ModuleLoadingMode::OnlyInterface ||
+      (Ctx.IgnoreAdjacentModules && ModuleID.Item.str() != STDLIB_NAME))
```

Comment:
So we still read from the module for the stdlib?

Reply:
Yes, we still allow the stdlib because too many tests rely on it using the swiftmodule built from full source. We'll likely want to change that behavior in the future as using the stdlib rebuilt from the swiftinterface provided results closer to what 3rd party devs see.